### PR TITLE
chore: update snafu to make clippy happy

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

In our another project which use greptimedb as a submodule, the snafu version is 0.8.4. However, the snafu version in greptimedb is 0.8.3. A [commit](https://github.com/shepmaster/snafu/commit/298b3c320c9b8aeb4521289651e74cc455c3d3d2) in snafu in between the two versions makes clippy runs fail in that project. So this PR updates the snafu in greptimedb and resolve the clippy issues.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
LGTM

**Comment by fengjiachun** [comment]
CI failed again cc @MichaelScofield

File: .github/workflows/develop.yml [hunk 1]
```diff
@@ -321,11 +321,6 @@ jobs:
       - uses: dtolnay/rust-toolchain@master
         with:
           toolchain: ${{ env.RUST_TOOLCHAIN }}
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2
-        with:
-          # Shares across multiple jobs
-          shared-key: "fuzz-test-targets"
```
Comment (WenyXu) [inline .github/workflows/develop.yml:328]: I think we should keep the cache

**Comment by MichaelScofield** [comment]
@WenyXu PTAL

**Review by WenyXu — APPROVED** [review]

**Pull request merged**